torch/csrc/Generator.cpp

```diff
@@ -79,6 +83,11 @@ static PyObject* THPGenerator_pynew(
         c10::DeviceTypeName(device.type()),
         " is not supported for torch.Generator() api.");
   }
+  self->weakreflist = NULL;
+
+  static py::handle _generator_registry = py::module::import("torch").attr("random").attr("_generator_registry");
+  _generator_registry.attr("add")(py::cast<py::object>((PyObject*)self.get()));
```

Comment:
move the add above. You don't want to do the "add" dynamically every time right?